Dockerfile

```diff
@@ -1,81 +1,30 @@
-FROM ruby:2.7.1-alpine3.10
+FROM quay.io/devto/ruby:2.7.1
 
-#------------------------------------------------------------------------------
-#
-# Install Project dependencies
-#
-#------------------------------------------------------------------------------
-RUN apk update -qq && apk add git nodejs postgresql-client ruby-dev build-base \
-  less libxml2-dev libxslt-dev pcre-dev libffi-dev postgresql-dev tzdata imagemagick \
-  libcurl curl-dev yarn
+USER root
 
-#------------------------------------------------------------------------------
-#
-# Install required bundler version
-#
-#------------------------------------------------------------------------------
-RUN gem install bundler:2.1.4
+RUN curl -sL https://dl.yarnpkg.com/rpm/yarn.repo -o /etc/yum.repos.d/yarn.repo && \
+    dnf install -y bash git nodejs postgresql ruby-devel less libxml2-devel libxslt-devel \
+    pcre-devel libffi-devel postgresql-devel tzdata ImageMagick libcurl \
+    libcurl-devel yarn
 
-#------------------------------------------------------------------------------
-#
-# Define working directory
-#
-#------------------------------------------------------------------------------
-WORKDIR /usr/src/app
+ENV BUNDLER_VERSION=2.1.4
+RUN gem install bundler:${BUNDLER_VERSION}
+
+RUN mkdir -p /opt/apps/devto
+WORKDIR /opt/apps/devto
 
-#------------------------------------------------------------------------------
-#
-# Copy Gemfile and run bundle install
-#
-#------------------------------------------------------------------------------
 COPY ./.ruby-version .
 COPY ./Gemfile ./Gemfile.lock ./
-RUN bundle install --jobs 20 --retry 5
+RUN bundle check || bundle install --jobs 20 --retry 5
 
-#------------------------------------------------------------------------------
-#
-# Copy Package.json and yarn.lock
-#
-#------------------------------------------------------------------------------
 COPY ./package.json ./yarn.lock ./.yarnrc ./
 COPY ./.yarn ./.yarn
 
-#------------------------------------------------------------------------------
-#
-# Install packages
-#
-#------------------------------------------------------------------------------
 RUN yarn install
 
-# timeout extension required to ensure
-# system work properly on first time load
-ENV RACK_TIMEOUT_WAIT_TIMEOUT=10000 \
-  RACK_TIMEOUT_SERVICE_TIMEOUT=10000 \
-  STATEMENT_TIMEOUT=10000
-
-# Run mode configuration between dev / demo
-# for entrypoint script behaviour
-ENV	RUN_MODE="demo"
```

Comment:
We were talking about completely removing this from the entry point script too at one point.

Reply:
We should make the Entrypoints do one thing, not many. That is why I removed it and we can figure out how to make demo mode a thing if it is a needed feature.